reporting_design.md

```diff
@@ -0,0 +1,193 @@
+Skt data flow and reporting design
+==================================
+
+Each skt stage command has:
+
+* Two inputs:
+    * Command-line options
+        * One of which may be specifying a configuration file containing
+          some command-line option values. The file cannot be used to
+          specify anything, which cannot be set with command-line options.
+          Command-line options following this option override values in
+          the file. This option can have a default, such as ~/.skt.yaml
+    * Input directory
+* Two outputs:
+    * Output directory
+    * Exit status summarizing the produced contents of the output
+      directory. I.e. the output directory contains all the data encoded
+      in the exit status.
+
+Among skt options are: input directory and output directory
+input directory by default is current directory
+output directory by default is input directory
+
+None of the different skt stage commands have output file names named the
+same, so they cannot step onto each other's toes. However, re-running the same
+skt stage command replaces existing files completely. E.g. it will need to
+remove all files that might have been created by the previous run of the same
+command, before starting adding its own, so the output file set is not
+inconsistent.
```

Comment:
I would prefer that we leave this to the administrator/operator/pipeline to clean up rather than make these applications aware that they need to remove or clear something. They should output whatever they need to without needing to know that something needs to be deleted.